src/services/CertificateGenerationService.ts

```diff
@@ -563,6 +566,39 @@ class CertificateGenerationService {
 
         return defectString;
     }
+    /**
+     * To fetch trailer registration
+     * @param vin The vin of the trailer
+     * @param make The make of the trailer
+     * @returns A payload containing the TRN of the trailer.
+     */
+     public async getTrailerRegistration(vin: string, make: string) {
+        const config: IInvokeConfig = this.config.getInvokeConfig();
+        const invokeParams: any = {
+            FunctionName: config.functions.trailerRegistration.name,
+            InvocationType: "RequestResponse",
+            LogType: "Tail",
+            Payload: JSON.stringify({
+                httpMethod: "GET",
+                path: `${Configuration.getInstance().getTrailerServiceVersion()}/trailers/${vin}?make=${make}`,
+                pathParameters: {
+                    proxy: `${Configuration.getInstance().getTrailerServiceVersion()}/trailers`
+                },
+                queryStringParameters: {
+                    make
+                }
+            }),
+        };
+        const response = await this.lambdaClient.invoke(invokeParams);
+        try {
+             const payload: any = this.lambdaClient.validateInvocationResponse(response);
+             const trailerRegistration = JSON.parse(payload.body) as ITrailerRegistration;
+             return trailerRegistration.trn;
+         } catch (e) {
+             console.log("Error on fetching trailer registration", e);
+             return undefined;
```

Comment:
Need to check this behaviour with the business

Reply:
@bristol-mike-jones any update on this?